Mention lowercase doctype in style guide (#38578)
mdn/content: files/en-us/mdn/writing_guidelines/writing_style_guide/code_style_guide/html/index.md

### Doctype

You should use the HTML5 {{Glossary("Doctype", "doctype")}}.

```html example-good
<!doctype html>
```

> > [!NOTE]  

Reviewer: Agree, no need to add this note here. The lower-case convention on MDN applies to constructs other than doctype and the "Casing" section explains that convention

> > [!NOTE]  

Reviewer: ```suggestion
```

> > The <!doctype html> declaration is case-insensitive and is commonly written as <!DOCTYPE html>.

Reviewer 2: Again don't say this. It doesn't matter in the style guide how it's "commonly written". What matters is what should be written on MDN. This is a prescriptive document, not descriptive.

> > The <!doctype html> declaration is case-insensitive and is commonly written as <!DOCTYPE html>.

Reviewer: ```suggestion
```

```html example-good
<!doctype html>
```
> [!NOTE]  

> > The doctype declaration (`<!doctype html>`) is case-insensitive. While MDN follows the convention of using lowercase, it is also commonly written as `<!DOCTYPE html>`.  

Reviewer 2: This is our style guide. You shouldn't need to say what else is okay; just say what is okay on MDN.

```suggestion
MDN always writes the doctype declaration in all-lowercase: `<!doctype html>`. This is consistent with our general convention to use lowercase where possible, such as HTML tags, attributes, or hexadecimal digits.
```

> > [!NOTE]  
> > The doctype declaration (`<!doctype html>`) is case-insensitive. While MDN follows the convention of using lowercase, it is also commonly written as `<!DOCTYPE html>`.  

Reviewer: Thanks @Josh-Cena for chiming in. Definitely, we need to emphasize the convention we use on MDN - so the second sentence from @Ipshita29 needs a little tweaking. But also, we do first need to state that the declaration is case-insensitive. This can be followed by the extra context you've provided for this convention.

I propose the following:

```suggestion
The doctype declaration is case-insensitive. On MDN, we follow the convention of writing it in lowercase. This is consistent with our general convention to use lowercase in case-insensitive contexts, such as HTML tags, attributes, and hexadecimal values.
```

> > The doctype declaration (`<!doctype html>`) is case-insensitive. While MDN follows the convention of using lowercase, it is also commonly written as `<!DOCTYPE html>`.  

Reviewer 2: IMO, we don't need to state facts and we should assume readers already know everything about the subject matter; no strong preferences though.

> > The doctype declaration (`<!doctype html>`) is case-insensitive. While MDN follows the convention of using lowercase, it is also commonly written as `<!DOCTYPE html>`.  

Author: Sure I will work on the following changes.

> > The doctype declaration (`<!doctype html>`) is case-insensitive. While MDN follows the convention of using lowercase, it is also commonly written as `<!DOCTYPE html>`.  

Author: I've made the requested changes.Please review and let me know if any further adjustments are needed. Thank you!

<input required="required" />
```

## Casing convention on MDN

> Use lowercase for all HTML tags, including the doctype declaration, element names, and attribute names/values, which are case-insensitive. This creates a consistent appearance and allows for faster markup writing. This convention applies to all case-insensitive contexts.

Reviewer 2: ```suggestion
Use lowercase for all case-insensitive constructs, including the doctype declaration, element names, and attribute names/values. This creates a consistent appearance and allows for faster markup writing.
```
